src-tauri/src/lib.rs

```diff
@@ -193,7 +215,42 @@ async fn do_install(nmod: &thermite::model::ModVersion, game_path: &std::path::P
     let download_path = format!("{}/{}", download_directory, filename);
     log::info!("Download path: {download_path}");
 
-    let nfile = thermite::core::manage::download_file(&nmod.url, download_path).unwrap();
+    let last_emit = RefCell::new(Instant::now()); // Keep track of the last time a signal was emitted
+    let nfile = thermite::core::manage::download_file_with_progress(
+        &nmod.url,
+        download_path,
+        |delta, current, total| {
+            if delta != 0 {
+                // Only emit a signal once every 250ms
+                let time_since_last_emit = Instant::now().duration_since(*last_emit.borrow());
+                if time_since_last_emit >= Duration::from_millis(250) {
```

Comment:
When I initially wrote it, it just printed out the progress to cli. When printing every update, I'd would slow down the download from 2sec on my 1gbps connection to 30-40sec as it would stall the download for every print.

I'm assuming pushing out an event everytime has similar performance impact. With Northstar being ~70MB it's 70k events when in reality we don't need to push out every single one but just need to update in a timely basis to give the impression of fluent progress.

Could reduce the latency between updates to e.g. 100ms if we want.

Ironically the faster the update rate, the slower the download speed will be xD

Reply:
I highly suspect the `console.log` invocations slowing down the UI (I don't see how download speed would be impacted, since actual downloading is done by the backend?).
Does thermite send an event for each downloaded kilobyte?

Reply:
> Does thermite send an event for each downloaded kilobyte?

`libthermite` does a callback for every KiB, yes. From there I can decide whether I actually want to fire an event or not ^^